src/range.rs

```diff
@@ -258,7 +260,7 @@ where
 {
     Range(
         z.clone(),
-        Box::new(move |sz| {
+        Rc::new(move |sz| {
```

Comment:
so what was the deal here? all the Box -> Rc

Reply:
A `Box` can only be cloned if the thing it's boxing is cloneable. When you clone an Rc you are simply incrementing the count. This is so I can actually hand functions around to things on different levels since I can't clone functions.